src/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -251,16 +251,36 @@ public void Name_CurrentlyExecutingAssembly()
             Assert.StartsWith("System.Reflection.Tests", assemblyName.Name);
         }
 
+        [Fact]
+        public static void FullName_WithPublicKey()
+        {
+            AssemblyName assemblyName = new AssemblyName("MyAssemblyName, Version=1.0.0.0");
+            assemblyName.SetPublicKeyToken(typeof(object).Assembly.GetName().GetPublicKeyToken());
```

Comment:
Calling SetPublicKeyToken() defeats the purpose of this test. You want to create an AssemblyName that has only a PublicKey.